Eastern-Research-Group/mywaterway
# Feature/missing assessments in gis

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3777026

## Main Changes:
* For locations on the Community page where the GIS does not return all possible assessments - we manually fetch the missing assessments using the ATTAINS assessments and assessmentUnits services. We then use this information to piece together the assessments and display them alongside the other assessments sans spatial data.

## Steps To Test:
The following HUC12s are known locations with this issue. Navigating to them on the Dev or Prod sites will result in some waterbodies not being displayed. 
```
010801040303
030502010707
040500030701
041504010306
071100060502
171003010802
170800050906
```

1. Navigate to some or all of the HUC12s in the list above on the Community page. Verify that:
* The correct number of waterbodies is displayed on the Overview tab' Waterbody count. On the dev/prod sites there is an issue where this number is sometimes incorrect due to some assessments being missing from the GIS services.
*  A message is displayed above the accordion list, "Some waterbodies are not visible on the map."
* The generated waterbodies are displayed on all tabs that are relevant to them based on their use statuses, (Swimming, Eating Fish, etc.)
* Waterbodies in the list have all possible information including Evaluated Uses and Impairments (if applicable) and a working link to the Waterbody Report page.
* The "View on Map" button is displayed but disabled with a message below it.

2. Verify the HTTP Intercept Cypress test passes.
3. If any step of the process fails due to results being unexpectedly empty or a service error occurs this is treated as if one of the Points/Lines/Areas services fails to prevent incomplete data from being displayed. This can be verified by blocking the ATTAINs assessmentUnits service in the network tab (https://attains.epa.gov/attains-public/api/assessmentUnits).
4. Verify normal locations like Washington DC are unaffected by these changes.
